RecoPPS/Local/interface/TimingRecHitProducerAlgorithm.h

```diff
@@ -0,0 +1,46 @@
+/****************************************************************************
+*
+* This is a part of PPS-TOTEM offline software.
+* Authors:
+*   Laurent Forthomme (laurent.forthomme@cern.ch)
+*
+****************************************************************************/
+
+#ifndef RecoPPS_Local_TimingRecHitProducerAlgorithm
+#define RecoPPS_Local_TimingRecHitProducerAlgorithm
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/Utils/interface/FormulaEvaluator.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "CondFormats/PPSObjects/interface/PPSTimingCalibration.h"
+#include "CondFormats/PPSObjects/interface/PPSTimingCalibrationLUT.h"
+
+template <typename G, typename D, typename R>
+class TimingRecHitProducerAlgorithm {
+public:
+  explicit TimingRecHitProducerAlgorithm(const edm::ParameterSet& iConfig)
+      : ts_to_ns_(iConfig.getParameter<double>("timeSliceNs")),
+        apply_calib_(iConfig.getParameter<bool>("applyCalibration")) {}
+  virtual ~TimingRecHitProducerAlgorithm() = default;
+
+  void setCalibration(const PPSTimingCalibration& calib, const PPSTimingCalibrationLUT& calibLUT) {
+    calib_ = calib;
+    calibLUT_ = calibLUT;
```

Comment:
Good point. Although this method was used all along run 2 for the PPS diamond reconstruction scheme. Surely this can be optimised a bit (even though the objects are not that large, and the number of channels remain limited...)

Reply:
Now I realise I forgot to address this in my last push. Will come with a follow-up in a short while. In the meantime the latest additions can already be tested.

Reply:
Done with 2e0936e8bef.